src/System.Private.CoreLib/shared/System/Diagnostics/Debug.cs

```diff
@@ -346,5 +339,35 @@ private sealed class DebugAssertException : Exception
         internal static Action<string, string, string, string> s_ShowDialog = ShowDialog;
```

Comment:
Do we still need the `s_ShowDialog` and `s_WriteCore` delegates?

Reply:
After the SetProvider API gets API approved I can remove them

Reply:
Why does it depend on SetProvider API gets API approved? These delegates are internal implementation details just like the SetProvider API.

Reply:
My understanding is because the unit tests can't inherit from a class that doesn't have a public contract. So in order for the tests to "mock" out the behavior, they still need to use reflection, and set these delegates.

Reply:
> the unit tests can't inherit from a class 

They can (e.g. System.Linq.Expressions tests do that). It is just something we try to avoid in most cases.